alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/settings/DestinationMailSettings.kt

```diff
@@ -0,0 +1,68 @@
+package com.amazon.opendistroforelasticsearch.alerting.settings
+
+import org.elasticsearch.common.settings.Setting
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.settings.SecureSetting
+import org.elasticsearch.common.settings.SecureString
+import java.io.IOException
+
+/**
+ * settings specific to mail destination.
+ */
+data class DestinationMailSettings(
+    val host: String,
+    val port: Int,
+    val method: String,
+    val from: String,
+    val username: SecureString,
+    val password: SecureString
+) {
+    companion object {
+        val DESTINATION_MAIL_HOST = Setting.simpleString(
+                "opendistro.alerting.destination.mail.host",
+                "localhost",
```

Comment:
After doing some manual testing, I noticed that these default settings like `localhost` for the SMTP host and `opendistro-alerting@localhost` for the from email don't work.

Ideally we should choose good defaults that work out of the box and if we are unable to do so, the settings should have no defaults and instead be required.